### PR TITLE
DH-15

### PR DESCRIPTION
**Pull request by Dyrk84**
Reworked TextSeparator
New characters variables: accuracy, finalAccuracy and defense.
New characters method: accuracyCalculation().
Renamed and reworked attackInitiating() method to battle() method in Fight class.
2 new method in Fight class: monsterAccuracy() and championAccuracy().
Reworked printed texts.

### CODE REVIEW

File: src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java [hunk 1]
```diff
@@ -6,13 +6,25 @@
...
+    int getAccuracy();
+    void setAccuracy(int accuracy);
+    int getDefense();
+    void setDefense(int defense);
+    int getFinalAccuracy();
+    void accuracyCalculation();
+    int getNumOfDices();
+    void setNumOfDices(int numOfDices);
+    int getMaxDamage();
+    void setMaxDamage(int maxDamage);
```
Comment (gergely-szanto-epam) [inline src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java]: Please delete unused interface methods (all that are grey)

File: src/main/java/hu/dungeonhunter/characters/monsters/Goblin.java [hunk 2]
```diff
@@ -8,22 +9,35 @@
 
     @Setter
     @Getter
-    private int hp;
+    private int finalAccuracy;
```
Comment (gergely-szanto-epam) [inline src/main/java/hu/dungeonhunter/characters/monsters/Goblin.java]: You're overthinking and over using class variables. Final accuracy comes from a calculation based on the base accuracy. You don't have to store this calculation result here, you need to pass the result in the related method.
The same goes for finalInitiation. Please remove these from all the character classes and make the calculator methods return the result for Fight class.

File: src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java (hunk 1 above)
Comment (gergely-szanto-epam) [reply src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java]: Once you have deleted finalAccuracy and finalInitiation variables, you can remove their setters and getters from the interface as well.

File: src/main/java/hu/dungeonhunter/characters/champion/Champion.java [hunk 3]
```diff
@@ -89,10 +119,17 @@ public void drinkAHealingPotion() {
...
+        System.out.print("Champion initiation calculation: " + getInitiative() + " + ");
         int champInitRoll = Dice.rollDice(10, 1);
         System.out.println("Final Champion initiation: " + (getInitiative() + champInitRoll));
-        finalChampionInitiation = getInitiative() + champInitRoll;
+        finalInitiation = getInitiative() + champInitRoll;
+    }
+
+    public void accuracyCalculation() {
+        System.out.print("Champion accuracy calculation: " + getAccuracy() + " + ");
+        int AccuracyRoll = Dice.rollDice(100, 1);
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/characters/champion/Champion.java:102]: 1-100 roll for accuracy? Someone must be drunk :D

Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/characters/champion/Champion.java:102]: Base accuracy +1d100 if higher like enemy defense, the hit will do damage. Its ok.

File: src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java (hunk 1 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/characters/monsters/MonstersInterface.java]: I did it.

File: src/main/java/hu/dungeonhunter/characters/monsters/Goblin.java (hunk 2 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/characters/monsters/Goblin.java]: I wanted to use this setter for a test, but in the end I didn’t use it and forgot there. Thank you, i deleted it.

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 4]
```diff
@@ -12,408 +12,499 @@
...
+    private MonsterFactory monsterFactory = new MonsterFactory();
+    private Fight fight = new Fight();
+    private Champion champion = new Champion();
 
-        // test action (When)
-        fight.championAttack();
+    //segítő változók
+    int toTestMonsterCounter = fight.getMonsterCounter();
+    int toTestKilledMonsterCounter = monsterFactory.getKilledMonsterCounter();
+    int toTestHealingPotionCounter = champion.getHealingPotionCounter();
```
Comment (gergely-szanto-epam) [inline src/test/java/hu/dungeonhunter/TestFight.java]: Please rename these to indicate these are the initialized values. e.g.: startingMonsterCounter

Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java]: I did it, ty!

**Review by gergelyszanto — APPROVED** [review]

**Pull request merged**